sdk/src/client/api/block_builder/input_selection/mod.rs

```diff
@@ -38,80 +44,101 @@ pub struct InputSelection {
     required_inputs: HashSet<OutputId>,
     forbidden_inputs: HashSet<OutputId>,
     selected_inputs: Vec<InputSigningData>,
+    context_inputs: HashSet<ContextInput>,
     outputs: Vec<Output>,
     addresses: HashSet<Address>,
     burn: Option<Burn>,
     remainder_address: Option<Address>,
-    protocol_parameters: ProtocolParameters,
-    slot_index: SlotIndex,
+    creation_slot: SlotIndex,
+    latest_slot_commitment_id: SlotCommitmentId,
     requirements: Vec<Requirement>,
     automatically_transitioned: HashSet<ChainId>,
-    mana_allotments: u64,
+    auto_mana_allotment: Option<AutoManaAllotment>,
+    mana_allotments: BTreeMap<AccountId, u64>,
+    required_allotment_mana: u64,
     mana_rewards: HashMap<OutputId, u64>,
+    payload: Option<TaggedDataPayload>,
+    allow_additional_input_selection: bool,
+    transaction_capabilities: TransactionCapabilities,
+    protocol_parameters: ProtocolParameters,
 }
 
-/// Result of the input selection algorithm.
-#[derive(Clone, Debug)]
-pub struct Selected {
-    /// Selected inputs.
-    pub inputs: Vec<InputSigningData>,
-    /// Provided and created outputs.
-    pub outputs: Vec<Output>,
-    /// Remainder outputs information.
-    pub remainders: Vec<RemainderData>,
-    /// Mana rewards by input.
-    pub mana_rewards: HashMap<OutputId, u64>,
+/// Account and RMC for automatic mana allotment
+#[derive(Copy, Clone, Debug)]
+pub(crate) struct AutoManaAllotment {
```

Comment:
just to make sure... we still want to refer to this functionality as "auto mana allotment"? I thought that term wasn't really liked and something like "minimum mana allotment" was the preferred term?

Reply:
I'm not super opinionated but since the node team doesn't really like it because it's something different for them, we probably shouldn't create any ambiguity.

Reply:
okie dokie